dotnetv3/Bedrock-runtime/Scenarios/ConverseToolScenario/ConverseToolScenario.cs

```diff
@@ -0,0 +1,358 @@
+﻿// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved. 
+// SPDX-License-Identifier: Apache-2.0
+
+// snippet-start:[Bedrock.ConverseTool.dotnetv3.Scenario]
+
+using Amazon.BedrockRuntime;
+using Amazon.BedrockRuntime.Model;
+using Amazon.Runtime.Documents;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Console;
+
+namespace ConverseToolScenario;
+
+public static class ConverseToolScenario
+{
+    /*
+    Before running this .NET code example, set up your development environment, including your credentials.
+
+    This demo illustrates a tool use scenario using Amazon Bedrock's Converse API and a weather tool.
+    The script interacts with a foundation model on Amazon Bedrock to provide weather information based on user
+    input. It uses the Open-Meteo API (https://open-meteo.com) to retrieve current weather data for a given location.
+   */
+
+    public static BedrockActionsWrapper _bedrockActionsWrapper = null!;
+    public static WeatherTool _weatherTool = null!;
+    public static bool _interactive = true;
+
+    private static string model_id = "anthropic.claude-3-sonnet-20240229-v1:0";
+
+    private static string system_prompt = @"
+        You are a weather assistant that provides current weather data for user-specified locations using only
+        the Weather_Tool, which expects latitude and longitude. Infer the coordinates from the location yourself.
+        If the user provides coordinates, infer the approximate location and refer to it in your response.
+        To use the tool, you strictly apply the provided tool specification.
+
+        - Explain your step-by-step process, and give brief updates before each step.
+        - Only use the Weather_Tool for data. Never guess or make up information. 
+        - Repeat the tool use for subsequent requests if necessary.
+        - If the tool errors, apologize, explain weather is unavailable, and suggest other options.
+        - Report temperatures in °C (°F) and wind in km/h (mph). Keep weather reports concise. Sparingly use
+          emojis where appropriate.
+        - Only respond to weather queries. Remind off-topic users of your purpose. 
+        - Never claim to search online, access external data, or use tools besides Weather_Tool.
+        - Complete the entire process until you have all required data before sending the complete response.
+    "
+    ;
+
```

Comment:
I like off topic responses. For example, when asked what is snow - it responded:

I do not actually have enough information to explain what snow is, since I am an AI assistant focused solely on providing current weather reports using the specified Weather_Tool. As an AI, I do not have general knowledge about different weather phenomena beyond what is provided by the tool's data.